client/src/pages/Report/ApplicationSection/ApplicationSection.tsx

```diff
@@ -138,16 +117,22 @@ const ApplicationSection = () => {
         <SectionComponent
             icon={<SVGIcon iconName="application-icon" />}
             title={'Applications'}
-            callback={handleAddClick}
-        >
+            callback={() => setShowModal(true)}>
             {showModal && (
-                <OverlayComponent isDisplayed={showModal} onClose={handleCloseModal}>
-                    <p>No applications here (probably Wojciech dropped all of them)</p>
+                <OverlayComponent
+                    isDisplayed={showModal}
+                    onClose={handleCloseModal}
```

Comment:
Consider removing onClose button from the components interface as it's pretty specific